oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/MergeRunner.java

```diff
@@ -297,14 +332,20 @@ private class Task implements Callable<File> {
             mergeTaskPhaser.register();
         }
 
+
         @Override
-        public File call() {
+        public File call() throws Exception {
```

Comment:
Instead of throwing "Exception", a more specific exception should be thrown. There are two options here:
(a) use you own exception class, or
(b) use an existing one. 

I think appropriate would be java.io.EOFException (end-of-file): https://docs.oracle.com/javase/9/docs/api/java/io/EOFException.html

Reply:
I wonder, do we have a test case that checks an exception is thrown at the right time?

Reply:
I rebase https://github.com/apache/jackrabbit-oak/pull/536 on top of this PR, since the unit tests of these two kind of go together and cannot be tested without another one (both are bugs, so want to have unit test that is checking the right state). So the same comment is addressed there as well. Feel free to close this one and review the other.